### ISSUE TITLE
training returns NaN when batch size = 1

### ISSUE THREAD

**Issue by chenlingantelope** [YosefLab/scVI#681]
Due to some filtering I was doing on my dataset, some of the technical batches ended up having only 1 cell, and I think that might have messed up the dimensions of the matrices? I got this error consistently if any of the batches has only 1 cell, and the problem goes away when I remove those batches. It might be worth outputting an error to let people know that that is an issue and remove the batches with only 1 cell. 

[2020-06-06 12:57:25,373] WARNING - scvi.inference.trainer | Model training loss was NaN

**Comment by galenxing** [comment]
Hi @chenlingantelope , 

Would you be able to provide some code to reproduce this? Do you know how many batches you have with only a single cell?

**Comment by adamgayoso** [comment]
I think we would need to systematically test at what limit this becomes a problem. Though a temporary solution would be to provide a warning if a single batch (experimental kind) has fewer than 100 cells. @galenxing we can add this to `setup_anndata`

**Comment by adamgayoso** [comment]
@galenxing we should refactor the new `setup_anndata` lightly to finish with a check on the properties of the data, including checking this.

**Comment by adamgayoso** [comment]
Closing this as the warning has been provided in our new setup_anndata function in #708
